src/UnikraftExtension.ts

```diff
@@ -40,154 +50,171 @@ export class UnikraftExtension {
         this.context.subscriptions.push(externalLibrariesTreeView);
         this.context.subscriptions.push(this.kraftStatusBarItem);
         this.kraftStatusBarItem.show();
-    
-        await this.setupExtension();
-        this.registerCommands();
-    
+
+
         vscode.window.onDidCloseTerminal(t => {
             if (t.exitStatus) {
+                if (t.name === 'Kraft setup') {
+                    this.kraftChannel.appendLine('Installed kraft.');
+                    this.kraftStatusBarItem.text = 'Installed kraft.';
+                    this.kraftChannel.appendLine('Executing "kraft pkg update"...');
+                    this.kraftChannel.appendLine(execSync(
+                        'kraft pkg update',
+                        { env: env }
+                    ).toString()
+                    );
+                    this.kraftChannel.appendLine('Executed "kraft pkg update".');
+                    this.initExtension()
+                }
+
                 if (t.name === 'kraft run') {
                     this.kraftStatusBarItem.text = 'Done running project.';
                 } else if (t.name === 'kraft menuconfig') {
                     this.kraftStatusBarItem.text = 'Done configuring project.';
 
                     const errFile = '/tmp/err_kraft_configure';
-                    if (fs.existsSync(errFile)) {
-                        vscode.window.showErrorMessage(fs.readFileSync(errFile).toString());
+                    if (existsSync(errFile)) {
+                        vscode.window.showErrorMessage(readFileSync(errFile).toString());
 
-                        fs.rmSync(errFile);
+                        rmSync(errFile);
                     }
                 }
-    
+
                 if (!t.exitStatus.code) {
                     return;
                 }
-    
+
                 this.kraftChannel.appendLine(
                     `${t.name} exit code: ${t.exitStatus.code}`
                 );
             }
         });
 
         vscode.workspace.onDidSaveTextDocument((document: vscode.TextDocument) => {
-            if (path.basename(document.fileName) === 'kraft.yaml') {
-                const projectPath = utils.getProjectPath();
+            if (basename(document.fileName) === 'kraft.yaml') {
+                const projectPath = getProjectPath();
                 setupPythonSupport(projectPath);
                 reloadIncludes(projectPath);
                 reloadConfig(projectPath);
             }
 
-            if (path.basename(document.fileName) === '.config') {
-                reloadConfig(utils.getProjectPath());
+            if (basename(document.fileName) === '.config') {
+                reloadConfig(getProjectPath());
```

Comment:
The `.config` file can differ based on the target in `kraft.yaml`

It's usually `.config.<name>_<plat>-<arch>`

Reply:
Done 👍🏽